src/analysis/inner_product_space/pi_L2.lean

```diff
@@ -381,3 +381,86 @@ def linear_isometry_equiv.from_orthogonal_span_singleton
   (n : ℕ) [fact (finrank 𝕜 E = n + 1)] {v : E} (hv : v ≠ 0) :
   (𝕜 ∙ v)ᗮ ≃ₗᵢ[𝕜] (euclidean_space 𝕜 (fin n)) :=
 linear_isometry_equiv.of_inner_product_space (finrank_orthogonal_span_singleton hv)
+
+section complex
+
+variables {V : Type*}
+[inner_product_space 𝕜 V]
+[finite_dimensional 𝕜 V]
+
+variables {S : submodule 𝕜 V} {L : S →ₗᵢ[𝕜] V}
+
+instance S_complete : complete_space S := finite_dimensional.complete 𝕜 S
+
+instance V_complete : complete_space V := finite_dimensional.complete 𝕜 V
+
+open finite_dimensional
+
+/-- Let `S` be a subspace of a finite-dimensional complex inner product space `V`.  A linear
+isometry mapping `S` into `V` can be extended to a full isometry of `V`. -/
+noncomputable def linear_isometry.extend (L : S →ₗᵢ[𝕜] V): V →ₗᵢ[𝕜] V :=
+begin
+  -- Build an isometry from Sᗮ to L(S)ᗮ through euclidean_space
+  let d := finrank 𝕜 Sᗮ,
+  have dim_S : finrank 𝕜 Sᗮ = d := rfl,
```

Comment:
```lean
  have dim_S_perp : finrank 𝕜 Sᗮ = d := rfl,
```
would match the `dim_LS_perp` name below

Reply:
Good point